### PR TITLE
mcap_ros2: Fix serde of empty messages

### LINKED ISSUE #943: Mcap Decoder is not Capable of Decoding "std_msgs/msg/Empty"
**Description**
`mcap` decoder can't handle "std_msgs/msg/Empty" type and will throw exception.

- Version: `mcap==1.1.0`
- Platform: This was on ros-iron image with the `mcap==1.1.0` and `mcap-ros2-support==0.5.0` installed (both the latest releases)

**Steps To Reproduce**
1. Record a bag containing a "std_msgs/msg/Empty" message in MCAP format; either by running `ros2 bag record -a -s mcap -o empty_message_bag` and `ros2 topic pub empty_msg std_msgs/msg/Empty`, or by using Write Messages example in https://mcap.dev/docs/python/ros2_example#writing-messages.
2. Attempt to read all messages using the official Reading Messages example in https://mcap.dev/docs/python/ros2_noenv_example#reading-messages

**Expected Behavior**
mcap decoder can decode "std_msgs/msg/Empty" in the same way that the official rosbag2_py SequentialReader can do.

**Actual Behavior**
mcap decoder raises an exception when reading from a bag that contains a "std_msgs/msg/Empty" message.

```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/root/ubuntu/robolog/ros2/ros2_bag_reader.py", line 274, in messages
    message = next(self._messages)
  File "/root/ubuntu/.venv/lib/python3.10/site-packages/mcap/reader.py", line 194, in iter_decoded_messages
    schema, channel, message, decoded_message(schema, channel, message)
  File "/root/ubuntu/.venv/lib/python3.10/site-packages/mcap/reader.py", line 182, in decoded_message
    decoder = factory.decoder_for(channel.message_encoding, schema)
  File "/root/ubuntu/.venv/lib/python3.10/site-packages/mcap_ros2/decoder.py", line 45, in decoder_for
    raise McapROS2DecodeError(f'schema parsing failed for "{schema.name}"')
mcap_ros2.decoder.McapROS2DecodeError: schema parsing failed for "std_msgs/msg/Empty"
```

### PR DESCRIPTION
**Pull request by achim-k**
### Public-Facing Changes

mcap_ros2: Fix serde of empty messages

### Description
In case a message definition definition is empty, ROS 2 adds a `uint8 structure_needs_at_least_one_member` field when converting to IDL, to satisfy the requirement from IDL of not being empty. See also
https://design.ros2.org/articles/legacy_interface_definition.html

This patch inserts or skips that extra member when serializing or deserializing respectively.

This PR is pretty much equivalent to https://github.com/foxglove/rosmsg2-serialization/pull/16

Fixes #943

### CODE REVIEW

File: python/mcap-ros2-support/tests/generate.py [hunk 1]
```diff
@@ -33,15 +46,34 @@ def generate_sample_data():
     )
 
     for i in range(10):
-        s = String(data=f"string message {i}")
+        e = String(data=f"string message {i}")
```
Comment (jtbandes) [inline python/mcap-ros2-support/tests/generate.py:49]: maybe keep `s` or use a more clear name like `string` or `msg`?

File: python/mcap-ros2-support/mcap_ros2/_dynamic.py [hunk 2]
```diff
@@ -501,12 +516,14 @@ def _for_each_msgdef(
 ) -> None:
     cur_schema_name = schema_name
 
+    # Remove empty lines
+    schema_text = os.linesep.join([s for s in schema_text.splitlines() if s.strip()])
```
Comment (jtbandes) [inline python/mcap-ros2-support/mcap_ros2/_dynamic.py:520]: what's the purpose of this change?

**Review by jtbandes — APPROVED** [review]

**Comment by jtbandes** [comment]
Might also want to bump the version in this PR

File: python/mcap-ros2-support/mcap_ros2/_dynamic.py (hunk 2 above)
Comment (achim-k) [reply python/mcap-ros2-support/mcap_ros2/_dynamic.py:520]: This avoids the 
```py
        if not cur_schema_text:	
            continue
```
which was removed below. Even if our schema is empty, we still want to call `parse_message_string(...)`. 

What I could alternatively, is to add an if before the loop:
```py
if not schema_text.strip():
   # Empty message definition
   ...
   msgdef = parse_message_string(pkg_name, msg_name, cur_schema_text)
   fn(cur_schema_name, short_name, msgdef)
   return
```

**Review by jtbandes — APPROVED** [review]
